src/qhttp/StaticContent.cc

```diff
@@ -118,6 +78,14 @@ void StaticContent::add(
             requestPath /= "index.htm";
         }
 
+        // Handle the oft-expected 404 case here explicitly, rather than as an exception.
+        if (!fs::exists(requestPath)) {
```

Comment:
Not true -- see documentation for `fs::weakly_canonical` for details.  See also unit test which tests this case explicitly.

Reply:
My comment was based on what I found in https://www.boost.org/doc/libs/1_68_0/libs/filesystem/doc/reference.html#weakly_canonical which says:
```
Remarks: Uses operator/= to compose the returned path. Uses the status function to determine existence.
```

Reply:
From the documentation at the same place: 
> Returns: A path composed of the result of calling the canonical function on a path composed of the leading elements of p that exist, if any, followed by the elements of p that do not exist, if any.

Reply:
I've looked at the documentation for the function `status` that (according to the documentation) is invoked by `weakly_canonical`, and it looks like an absence of a path is not reported as an error. So, you're right by explicitly checking for the existence of a file.

One more small question in this context. The new code has 3 steps:
```
fs::weakly_canonical
if (fs::is_directory(requestPath)) { ...
if (!fs::exists(requestPath)) { ...
```
Would the second choke if the directory-like path (ending with `/`) didn't exist? Or, the result of the method is entirely based on a presence of `/` in the end of the path? The documentation is not very helpful here.

Reply:
Hmm, good catch!  I think this could throw in some circumstances as the code is currently written.  I'll test, tighten up if necessary, and add unit test coverage.

Reply:
Okay, I wrote some code to test this, and it turns out that `fs::is_directory(const path& p)` does _not_ throw if p does not exist; it returns `false`.

After testing I went back and reading carefully, and this is in fact documented, though somewhat indirectly. 

> `bool is_directory(const path& p);`
Returns: `is_directory(status(p))`... 

...and then looking up `status(p)` we have:

> Throws: `filesystem_error`. [Note: result values of `file_status(file_not_found)` and `file_status(type_unknown)` are not considered failures and do not cause an exception to be thrown. —end note]

Given this, the code on this PR is I believe currently correctly structured.  I did add a clarifying comment, though, because this is not obvious.